src/chaplin/lib/utils.coffee

```diff
@@ -2,11 +2,34 @@
 
 _ = require 'underscore'
 support = require 'chaplin/lib/support'
+ElementProto = if typeof Element != 'undefined' then Element.prototype else {}
+
 
 # Utilities
 # ---------
 
 utils =
+  # DOM Helpers
+  # --------------
+
+  matchesSelector: do ->
+    matches = ElementProto.matches ||
+      ElementProto.webkitMatchesSelector ||
+      ElementProto.mozMatchesSelector ||
+      ElementProto.msMatchesSelector ||
+      ElementProto.oMatchesSelector ||
+      #Make our own `Element#matches` for IE8
+      (selector) ->
+        #Use querySelectorAll to find all elements matching the selector,
+        #then check if the given element is included in that list.
+        #Executing the query on the parentNode reduces the resulting nodeList,
+        #(document doesn't have a parentNode).
+        nodeList = (this.parentNode || document).querySelectorAll(selector) || []
+        !!~indexOf(nodeList, this)
+
+      (node, selector) ->
+        matches.call(node, selector)
+
```

Comment:
In general I'm not a fan of functions wrapping a `call` just to preserve a signature. I think it's fine if in filterChildren we do `utils.matchesSelector.call node, selector` instead